autosklearn/ensembles/multiobjective_dummy_ensemble.py

```diff
@@ -0,0 +1,458 @@
+        -------
+        np.ndarray
+        """
+        return predictions[0]
+
+    def __str__(self) -> str:
+        return "Single Model:\n\tMembers: %s" "\n\tWeights: %s\n\tIdentifiers: %s" % (
+            self.indices_,
+            self.weights_,
+            " ".join(
+                [
+                    str(identifier)
+                    for idx, identifier in enumerate(self.identifiers_)
+                    if self.weights_[idx] > 0
+                ]
+            ),
+        )
+
+    def get_models_with_weights(
+        self, models: Dict[Tuple[int, int, float], BasePipeline]
+    ) -> List[Tuple[float, BasePipeline]]:
+        """Return a list of (weight, model) pairs for the model selected by this ensemble.
+
+        Parameters
+        ----------
+        models : dict {identifier : model object}
+            The identifiers are the same as the one presented to the fit()
+            method. Models can be used for nice printing.
+
+        Returns
+        -------
+        List[Tuple[float, BasePipeline]]
+        """
+        output = []
+        for i, weight in enumerate(self.weights_):
+            if weight > 0.0:
+                identifier = self.identifiers_[i]
+                model = models[identifier]
+                output.append((weight, model))
+
+        output.sort(reverse=True, key=lambda t: t[0])
+
+        return output
+
+    def get_identifiers_with_weights(
+        self,
+    ) -> List[Tuple[Tuple[int, int, float], float]]:
+        """Return a (identifier, weight)-pairs for the model selected by this ensemble.
+
+        Parameters
+        ----------
+        models : dict {identifier : model object}
+            The identifiers are the same as the one presented to the fit()
+            method. Models can be used for nice printing.
+
+        Returns
+        -------
+        List[Tuple[Tuple[int, int, float], float]
+        """
+        return list(zip(self.identifiers_, self.weights_))
+
+    def get_selected_model_identifiers(self) -> List[Tuple[int, int, float]]:
+        """Return identifier of models in the ensemble.
+
+        This includes models which have a weight of zero!
+
+        Returns
+        -------
+        list
+        """
+        output = []
+
+        for i, weight in enumerate(self.weights_):
+            identifier = self.identifiers_[i]
+            if weight > 0.0:
+                output.append(identifier)
+
+        return output
+
+    def get_validation_performance(self) -> float:
+        """Return validation performance of ensemble.
+
+        In case of multi-objective problem, only the first metric will be returned.
+
+        Return
+        ------
+        float
+        """
+        return self.best_model_score_
+
+
+class MultiObjectiveDummyEnsemble(AbstractMultiObjectiveEnsemble):
+    def __init__(
+        self,
+        task_type: int,
+        metrics: Sequence[Scorer] | Scorer,
+        random_state: Optional[Union[int, np.random.RandomState]],
```

Comment:
```suggestion
        random_state: int | np.random.RandomState | None,
```

Maybe default `= None` too?

Reply:
I applied `pyupgrade`, this should be fixed now. I don't think we should add a default here with the same argument as above.